src/test/regress/upgrade/citus_upgrade_test.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+"""citus_upgrade_test
+Usage:
+    citus_upgrade_test --bindir=<bindir> --citus-version=<citus-version> --pgxsdir=<pgxsdir> --pg-version=<pg-version> 
+
+Options:
+    --bindir=<bindir>                      The PostgreSQL executable directory(ex: '~/.pgenv/pgsql-10.4/bin')
+    --citus-version=<citus-version>        Citus version, this should be a branch or tag(ex: v8.0.0)
+    --pgxsdir=<pgxsdir>           	       Path to the PGXS directory(ex: ~/.pgenv/src/postgresql-11.3)
+    --pg-version<pg-version>               Major Postgres version to use(ex: 11)
+"""
+
+import subprocess
+import atexit
+import os
+
+import utils
+
+from docopt import docopt
+
+from config import (
+    CitusUpgradeConfig, NODE_PORTS, COORDINATOR_NAME, BEFORE_CITUS_UPGRADE_SCHEDULE,
+    NODE_NAMES, USER, AFTER_CITUS_UPGRADE_SCHEDULE, WORKER1PORT,
+    AFTER_CITUS_UPGRADE_COORD_SCHEDULE, BEFORE_CITUS_UPGRADE_COORD_SCHEDULE
+)
+from upgrade_common import initialize_temp_dir, initialize_citus_cluster, run_pg_regress, stop_databases
+
+
+def verify_initial_version(config):
+    for port in NODE_PORTS.values():
+        run_pg_regress(config.bindir, config.pg_srcdir,
+                       port, BEFORE_CITUS_UPGRADE_SCHEDULE.format(config.citus_version))
+
+
+def run_test_on_coordinator(config, schedule):
+    run_pg_regress(config.bindir, config.pg_srcdir,
+                   NODE_PORTS[COORDINATOR_NAME], schedule)
+
+
+def verify_upgrade_mixed_mode(config):
+    for port in NODE_PORTS.values():
+        schedule = AFTER_CITUS_UPGRADE_SCHEDULE
+        if port == WORKER1PORT:
+            schedule = BEFORE_CITUS_UPGRADE_SCHEDULE.format(
+                config.citus_version)
+        run_pg_regress(config.bindir, config.pg_srcdir,
+                       port, schedule)
+
+
+def verify_upgrade(config):
+    for port in NODE_PORTS.values():
+        run_pg_regress(config.bindir, config.pg_srcdir,
+                       port, AFTER_CITUS_UPGRADE_SCHEDULE)
+
+
+def install_citus(citus_version, pg_version):
+    with utils.cd('/'):
+        subprocess.call(
+            ['tar', 'xvf', './install-pg{}-citus{}.tar'.format(pg_version, citus_version)])
+
+def install_citus_master(pg_version):
```

Comment:
the functions `install_citus`, `install_citus_master`, `remove_citus_artifacts ` and `remove_citus_master_artifacts` seem to be sharing a lot of code. Can we reduce this to 2 functions:|
- `install_citus`
- `remove_citus`

that take a version argument where `master` is a valid version. We would just run a different path into the tar function extracting the files.